src/steps/index.test.ts

```diff
@@ -1,92 +1,107 @@
-import { createMockStepExecutionContext } from '@jupiterone/integration-sdk-testing';
+import {
+  createMockStepExecutionContext,
+  Recording,
+} from '@jupiterone/integration-sdk-testing';
+
+import { setupSigSciRecording } from '../../test/recording';
 
 import { IntegrationConfig } from '../config';
-import { buildGroupUserRelationships, fetchGroups, fetchUsers } from './access';
-import { fetchAccountDetails } from './account';
+import { fetchUsers } from './user';
+import { fetchCorps } from './organization';
 import { integrationConfig } from '../../test/config';
 
-test('should collect data', async () => {
-  const context = createMockStepExecutionContext<IntegrationConfig>({
-    instanceConfig: integrationConfig,
+describe('should collect corp and user data', () => {
+  let recording: Recording;
+
+  beforeEach(() => {
+    recording = setupSigSciRecording({
+      directory: __dirname,
+      name: 'buildCorpAndUserRelationship',
+    });
   });
 
-  // Simulates dependency graph execution.
-  // See https://github.com/JupiterOne/sdk/issues/262.
-  await fetchAccountDetails(context);
-  await fetchUsers(context);
-  await fetchGroups(context);
-  await buildGroupUserRelationships(context);
+  afterEach(async () => {
+    await recording.stop();
+  });
 
-  // Review snapshot, failure is a regression
-  expect({
-    numCollectedEntities: context.jobState.collectedEntities.length,
-    numCollectedRelationships: context.jobState.collectedRelationships.length,
-    collectedEntities: context.jobState.collectedEntities,
-    collectedRelationships: context.jobState.collectedRelationships,
-    encounteredTypes: context.jobState.encounteredTypes,
-  }).toMatchSnapshot();
+  test('validate client integration and converters', async () => {
+    // Arrange
+    const context = createMockStepExecutionContext<IntegrationConfig>({
+      instanceConfig: integrationConfig,
+    });
 
-  const accounts = context.jobState.collectedEntities.filter((e) =>
-    e._class.includes('Account'),
-  );
-  expect(accounts.length).toBeGreaterThan(0);
-  expect(accounts).toMatchGraphObjectSchema({
-    _class: ['Account'],
-    schema: {
-      additionalProperties: false,
-      properties: {
-        _type: { const: 'acme_account' },
-        manager: { type: 'string' },
-        _rawData: {
-          type: 'array',
-          items: { type: 'object' },
-        },
-      },
-      required: ['manager'],
-    },
-  });
+    // Act
+    await fetchCorps(context);
```

Comment:
If possible, I would attempt to isolate the tests for `fetchCorps` and `fetchUsers`, which will prevent you from having to re-record the entire test if you choose to add additional resources in the future.

Here's an example of an isolated test: https://github.com/JupiterOne/graph-aws/blob/main/src/steps/services/ec2/executionHandlers/vpcs.test.ts

Reply:
Cool, cool